modules/homa-lite/src/lib.rs

```diff
@@ -155,21 +180,51 @@ pub mod module {
 		///
 		/// Parameters:
 		/// - `amount`: The amount of Staking currency to be exchanged.
+		/// - `xcm_dest_weight`: The weight to be paid to the destination for the XCM transfer.
 		#[pallet::weight(< T as Config >::WeightInfo::request_mint())]
 		#[transactional]
-		pub fn request_mint(origin: OriginFor<T>, amount: Balance) -> DispatchResult {
-			let stash_account = Self::relay_chain_stash_account().ok_or(Error::<T>::RelayChainStashAccountNotSet)?;
-
+		pub fn request_mint(origin: OriginFor<T>, amount: Balance, xcm_dest_weight: Weight) -> DispatchResult {
 			let who = ensure_signed(origin)?;
-			let current_batch = Self::current_batch();
-			let staking_currency_id = T::StakingCurrencyId::get();
 
-			// TODO: Cross-chain transfer to the relay chain via XCM
-			T::Currency::transfer(staking_currency_id, &who, &stash_account, amount)?;
+			// Ensure the amount is above the minimum
+			ensure!(
+				amount >= T::MinimumMintThreshold::get(),
+				Error::<T>::MintAmountBelowMinimumThreshold
+			);
+
+			// Ensure the total amount staked doesn't exceed the cap.
+			let new_total_staked = Self::total_staked_amount()
+				.checked_add(amount)
+				.ok_or(ArithmeticError::Overflow)?;
+			let mint_cap = Self::staking_currency_mint_cap();
+			ensure!(
+				mint_cap.is_none() || new_total_staked <= mint_cap.unwrap(),
+				Error::<T>::ExceededStakingCurrencyMintCap
+			);
+
+			let current_batch = Self::current_batch();
+			let staking_currency = T::StakingCurrencyId::get();
+
+			// ensure the user has enough funds on their account.
+			T::Currency::ensure_can_withdraw(staking_currency, &who, amount)?;
+
+			// Cross-chain transfers the staking assets.
+			let xcm_result = T::CrossChainTransfer::transfer(
+				who.clone(),
+				staking_currency,
+				amount,
+				T::XcmSovereignSubAccount::get(),
+				xcm_dest_weight,
+			)?;
+			match xcm_result {
+				Outcome::Complete(_) => Ok(()),
+				_ => Err(Error::<T>::XCMTransferError),
+			}?;
```

Comment:
The readability could be improved:

```suggestion
			ensure!(matches!(xcm_result, Outcome::Complete(_)), Error::<T>::XCMTransferError);
```

More like plain English :)